File: src/2678.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int countSeniors(char **details, int detailsSize) {
    int retVal = 0;

    int age;
    int i;
    for (i = 0; i < detailsSize; ++i) {
        age = 10 * (details[i][11] - '0') + (details[i][12] - '0');
        if (age > 60) {
            retVal++;
        }
    }

    return retVal;
}

int main(int argc, char **argv) {
#define MAX_SIZE (100)
    struct testCaseType {
        char *details[MAX_SIZE];
        int detailsSize;
    } testCase[] = {{{"7868190130M7522", "5303914400F9211", "9273338290F4010"}, 3},
                    {{"1313579440F2036", "2921522980M5644"}, 2}};
    int numberOfTestCase = sizeof(testCase) / sizeof(testCase[0]);
    /* Example
     *  Input: details = ["7868190130M7522","5303914400F9211","9273338290F4010"]
     *  Output: 2
     *
     *  Input: details = ["1313579440F2036","2921522980M5644"]
     *  Output: 0
     */

    int answer = 0;
    int i, j;
    for (i = 0; i < numberOfTestCase; ++i) {
        printf("Input: details = [");
        for (j = 0; j < testCase[i].detailsSize; ++j) {
            printf("%s\"%s\"", (j == 0) ? "" : ",", testCase[i].details[j]);
        }
        printf("]\n");

        answer = countSeniors(testCase[i].details, testCase[i].detailsSize);
        printf("Output: %d\n", answer);

        printf("\n");
    }

    return EXIT_SUCCESS;
}

File: src/2678.py
import logging
import sys
from bisect import bisect_left, bisect_right
from collections import Counter, defaultdict, deque
from functools import cache, lru_cache
from heapq import heapify, heappop, heappush
from itertools import accumulate
from typing import List, Optional


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def countSeniors(self, details: List[str]) -> int:
        retVal = 0

        for detail in details:
            age = 10 * (ord(detail[11])-ord('0')) + (ord(detail[12])-ord('0'))
            if age > 60:
                retVal += 1

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for details in [["7868190130M7522", "5303914400F9211", "9273338290F4010"], ["1313579440F2036", "2921522980M5644"]]:
            # /* Example
            #  *  Input: details = ["7868190130M7522","5303914400F9211","9273338290F4010"]
            #  *  Output: 2
            #  *
            #  *  Input: details = ["1313579440F2036","2921522980M5644"]
            #  *  Output: 0
            #  */
            logging.info("Input: details = %s", details)

            retVal = pSolution.countSeniors(details)
            logging.info("Output: %d", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)
